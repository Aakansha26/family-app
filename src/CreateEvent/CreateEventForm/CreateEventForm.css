.createEventForm {
    flex: 1;
    padding: 50px 50px;
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 75vh;
  }

.createEventForm form {
    margin: 30px;
    width: 80%;
}

.form_field {
    margin: 13px;
}

