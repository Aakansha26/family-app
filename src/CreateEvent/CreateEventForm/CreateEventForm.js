import React, {useState} from 'react'
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import './CreateEventForm.css'
import db from '../../firebase'
import Snackbar from '@material-ui/core/Snackbar';
import Alert from '@material-ui/lab/Alert';

function CreateEventForm() {
    const [eventName, setEventName] = useState('')
    const [eventText, setEventText] = useState('')
    const [eventDate, setEventDate] = useState('')
    const [successSnackbarOpen, setSuccessSnackbarOpen] = useState(false);
    const [errorSnackbarOpen, setErrorSnackbarOpen] = useState(false);

    const handleNameChange = (event) => {
        setEventName(event.target.value);
    };

    const handleTextChange = (event) => {
        setEventText(event.target.value)
    };

    const handleDateChange = (event) => {
        setEventDate(event.target.value)
    };
  
    const invalidEventName = () => {
        if(eventName.length <= 0)
         return true;
        return false;
    }

    const handleSubmit = (event) => {
       event.preventDefault();
       if(invalidEventName())
       {
           openErrorSnackbar();
       }
       else
       {
            db.collection('events').add({
                    eventName: eventName,
                    eventDescription: eventText,
                    eventDate: eventDate
            })
            openSuccessSnackbar();
       }
      
       setEventName('');
       setEventText('');
       setEventDate('');
     
    }

    const openSuccessSnackbar = () => {
        setSuccessSnackbarOpen(true);
    }

    const closeSuccessSnackbar = () => {
        setSuccessSnackbarOpen(false);
    }

    const openErrorSnackbar = (message) => {
        setErrorSnackbarOpen(true);
        
    }

    const closeErrorSnackbar = () => {
        setErrorSnackbarOpen(false);
    }

    return (
        <div className="createEventForm">
            <h2>Time to Remember Family Occassions!</h2>
            <form>
                <div className="form_field">
                  <TextField 
                  id="filled-basic"
                  inputProps={{ "data-testid": "eventName-id" }} 
                  label="Event Name" 
                  variant="filled" 
                  value={eventName} 
                  onChange={handleNameChange} 
                  required />
                </div>

                <div className="form_field">
                    <TextField
                    id="filled-multiline-flexible"
                    inputProps={{ "data-testid": "eventDescription-id" }}
                    label="Event Description"
                    multiline
                    fullWidth
                    rows={3}
                    value={eventText}
                    onChange={handleTextChange}
                    variant="filled"
                    /> 
                </div>

                <div className="form_field">
                  <TextField
                        id="date"
                        inputProps={{ "data-testid": "eventDate-id" }}
                        label="Event Date"
                        variant="filled"
                        type="date"
                        value={eventDate}
                        required
                        onChange={handleDateChange}
                        InputLabelProps={{
                        shrink: true,
                        }}
                  />
                </div>
                <div className="form_field">
                    <Button variant="contained" color="secondary" onClick={handleSubmit}> Add Event
                    </Button>
                </div>
            </form>

            <Snackbar anchorOrigin={{
                    vertical: 'top',
                    horizontal: 'right',
                    }}
                    open={successSnackbarOpen}
                    autoHideDuration={4000}
                    onClose={closeSuccessSnackbar}>
                    <Alert elevation={6} variant="filled" onClose={closeSuccessSnackbar} severity="success">
                        Event Added Successfully!
                    </Alert>
            </Snackbar>

            <Snackbar anchorOrigin={{
                    vertical: 'top',
                    horizontal: 'right',
                    }}
                    open={errorSnackbarOpen}
                    autoHideDuration={4000}
                    onClose={closeErrorSnackbar}>
                    <Alert elevation={6} variant="filled" onClose={closeErrorSnackbar} severity="error">
                        Event Name cannot be empty
                    </Alert>
            </Snackbar>

        </div>
    )
}

export default CreateEventForm