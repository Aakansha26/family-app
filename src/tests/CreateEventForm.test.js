import React from 'react'
import CreateEventForm from '../CreateEvent/CreateEventForm/CreateEventForm'
import renderer from 'react-test-renderer'
import { configure } from 'enzyme';
import React16Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new React16Adapter() });
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";


describe("CreateEventForm Component", () => {

    it("Snapshot test", () => {
        const tree = renderer.create(<CreateEventForm />);
        let snapshot = tree.toJSON();
        expect(snapshot).toMatchSnapshot();
    })

    it("CreateEventForm Renders properly", () => {
        render(<CreateEventForm />);
        expect(screen.getByText("Time to Remember Family Occassions!")).toBeInTheDocument();
        expect(screen.getByText("Event Name")).toBeInTheDocument();
        expect(screen.getByText("Event Description")).toBeInTheDocument();
        expect(screen.getByText("Event Date")).toBeInTheDocument();
        expect(screen.getByText("Add Event")).toBeInTheDocument();
    })

    it("Filling the Form", () => {
        const { container, getByText, getByTestId } = render(<CreateEventForm />);

        expect(getByText("Add Event")).toBeInTheDocument();

        fireEvent.change(getByTestId('eventName-id'), { target: { value: "Birthday" } })
        expect(getByTestId('eventName-id')).toHaveValue("Birthday");

        fireEvent.change(getByTestId('eventDescription-id'), { target: { value: "Buy birthday gift" } })
        expect(getByTestId('eventDescription-id')).toHaveValue("Buy birthday gift");


    })

})

describe("Test to check 'Add Event' button works properly", () => {
    it("when input fields are empty", () => {

     const { container, getByText } = render(<CreateEventForm />);
     fireEvent.click(getByText('Add Event'));

     expect(screen.getByText("Event Name cannot be empty")).toBeInTheDocument();

    });

    it("when input fields are not empty", () => {
        const { container, getByText, getByTestId } = render(<CreateEventForm />);
        fireEvent.change(getByTestId('eventName-id'), { target: { value: "Birthday" } })
        fireEvent.change(getByTestId('eventDescription-id'), { target: { value: "Buy birthday gift" } })
        fireEvent.change(getByTestId('eventDate-id'), { target: { value: '2020-05-24'  } })
        fireEvent.click(getByText('Add Event'));
        expect(screen.getByText("Event Added Successfully!")).toBeInTheDocument();
    })
})