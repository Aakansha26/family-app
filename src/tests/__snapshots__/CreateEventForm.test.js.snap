// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEventForm Component Snapshot test 1`] = `
<div
  className="createEventForm"
>
  <h2>
    Time to Remember Family Occassions!
  </h2>
  <form>
    <div
      className="form_field"
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-required Mui-required"
          data-shrink={false}
          htmlFor="filled-basic"
          id="filled-basic-label"
        >
          Event Name
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input"
            data-testid="eventName-id"
            disabled={false}
            id="filled-basic"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="form_field"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
          data-shrink={false}
          htmlFor="filled-multiline-flexible"
          id="filled-multiline-flexible-label"
        >
          Event Description
        </label>
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
            data-testid="eventDescription-id"
            disabled={false}
            id="filled-multiline-flexible"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            rows={3}
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="form_field"
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-required Mui-required"
          data-shrink={true}
          htmlFor="date"
          id="date-label"
        >
          Event Date
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input"
            data-testid="eventDate-id"
            disabled={false}
            id="date"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="date"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="form_field"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
           Add Event
        </span>
      </button>
    </div>
  </form>
</div>
`;
